File: Server/Models/TableConfiguration.cs
using System;
using System.Collections.Generic;

namespace Server.Models
{
  public class TableConfiguration
  {
    public TableConfiguration(int id)
    {
      this.Id = id;
      this.Table = new Table();
      this.Name = "";
    }
    public int Id { get; set; }
    public string Name { get; set; }
    public bool Active { get; set; }
    public Table Table { get; set; }
  }
  public class Table
  {
    public Table()
    {
      this.ColumnHeaders = new List<Header>();
      this.RowHeaders = new List<Header>();
      this.TableList = new List<TableList>();
    }
    public List<Header> ColumnHeaders { get; set; }
    public List<Header> RowHeaders { get; set; }
    public List<TableList> TableList { get; set; }

  }
  public class Header
  {
    public string Id { get; set; }
    public string Value { get; set; }
  }
  public class TableList
  {
    public TableList()
    {
      this.List = new List<string>();
    }
    public String UniqueId { get; set; }
    public string RId { get; set; }
    public string CId { get; set; }
    public List<String> List { get; set; }
  }

}

File: Client/src/app/Admin/shared/TableConfiguration.ts
import { ToolboxItem } from "../shared/ToolboxItem";


export class TableConfiguration {
  public Id: number;
  public Table: Table;
  public Name: string;
  public Active: boolean;
}


export class Table {
  constructor(tl: TableList[], rh: Header[], ch: Header[]) {
    this.TableList = tl;
    this.RowHeaders = rh;
    this.ColumnHeaders = ch;
  }

  public ColumnHeaders: Header[];
  public RowHeaders: Header[];
  public TableList: TableList[];


}

export class TableList {
  constructor(id: string, cid: string, rid, list: ToolboxItem[]) {
    this.RId = rid;
    this.CId = cid;
    this.List = list;
    this.UniqueId = id;
  }
  public UniqueId: string;
  public RId: string;
  public CId: string;
  public List: ToolboxItem[];
}

export class Header {
  constructor(id: string, value: string) {
    this.Id = id;
    this.Value = value;
  }
  public Id: string;
  public Value: string;
}

